src/csharp/Grpc.Tools/ProtoCompile.cs

```diff
@@ -278,6 +278,12 @@ public class ProtoCompile : ToolTask
         /// </summary>
         public string[] OutputOptions { get; set; }
 
+        /// <summary>
+        /// Additional options that will be passed directly to protoc.
+        /// For example, "experimental_allow_proto3_optional"
+        /// </summary>
+        public string[] AdditionalProtocOptions { get; set; }
```

Comment:
I'm open to alternative parameter name suggestions.

Reply:
Am also open to making the option explicit to `experimental_allow_proto3_optional` such as:

```
public bool ExperimentalAllowProto3Optional { get; set; }
```

The downside is that any future flags wouldn't be supported without a code update. The upside however is that it'd be safer (i.e. you couldn't inject arbitrary options)

Reply:
@jtattermusch I went ahead and renamed it to `AdditionalProtocArguments` and made it so that I didn't automatically prefix it.